usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedFareAttribute.java

```diff
@@ -0,0 +1,96 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.mobilitydata.gtfsvalidator.domain.entity.ParsedEntity;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.fareattributes.FareAttribute;
+import org.mobilitydata.gtfsvalidator.usecase.notice.error.EntityMustBeUniqueNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.error.MissingRequiredValueNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.error.UnexpectedValueNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsDataRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.sql.SQLIntegrityConstraintViolationException;
+
+public class ProcessParsedFareAttribute {
+    private final ValidationResultRepository resultRepository;
+    private final GtfsDataRepository gtfsDataRepository;
+    private final FareAttribute.FareAttributeBuilder builder;
+
+    public ProcessParsedFareAttribute(final ValidationResultRepository resultRepository,
+                                      final GtfsDataRepository gtfsDataRepository,
+                                      final FareAttribute.FareAttributeBuilder builder) {
+        this.resultRepository = resultRepository;
+        this.gtfsDataRepository = gtfsDataRepository;
+        this.builder = builder;
+    }
+
+    public void execute(final ParsedEntity validatedFareAttribute) throws IllegalArgumentException,
+            SQLIntegrityConstraintViolationException {
+
+        final String fareId = (String) validatedFareAttribute.get("fare_id");
+        final Float price = (Float) validatedFareAttribute.get("price");
+        final String currencyType = (String) validatedFareAttribute.get("currency_type");
+        final Integer paymentMethod = (Integer) validatedFareAttribute.get("payment_method");
+        final Integer transfers = (Integer) validatedFareAttribute.get("transfers");
+        final String agencyId = (String) validatedFareAttribute.get("agency_id");
+        final Integer transferDuration = (Integer) validatedFareAttribute.get("transfer_duration");
+
+        try {
+            builder.fareId(fareId)
+                    .price(price)
+                    .currencyType(currencyType)
+                    .paymentMethod(paymentMethod)
+                    .transfers(transfers)
+                    .agencyId(agencyId)
+                    .transferDuration(transferDuration);
+
+            gtfsDataRepository.addFareAttribute(builder.build());
+
+        } catch (IllegalArgumentException e) {
+
+            if (fareId == null) {
```

Comment:
This code would be a lot more efficient if we avoided throwing any exceptions, and just checked for these conditions before adding the object to the `gtfsDataRepository`.

Also, the more I see code that's expected to run in our validation process in a `catch` block, the less I like it. IMHO code that handles data validation is expected to run as part of our application under normal conditions and shouldn't be in `catch` blocks.

Reply:
I see your point regarding the exception, but it feels that checking before building would kinda defeat the point of having the builder, with the validation being done outside of it. If it were to be reused it would come with implicit expectations that you check the data as it would let you build anything. For the catch, I agree, the less we do in catch blocks, the better. In that case though it's a duplicate of what happens in the Builder class.
I proposed to @lionel-nj  to return a list of Notice instead of raising an exception. He pointed out that our entity classes are in the domain layer whereas our Notices are in the use case one, because of the visitor pattern used to export them (they all have an export method taking a NoticeExporter parameter, which also isn't in domain). Maybe we should focus on how to refactor so that notices can be in the domain as well, which you already mentionned would make more sense.

Reply:
An issued on how to refactor notice to be part of the domain layer has been opened. See https://github.com/MobilityData/gtfs-validator/issues/122

Reply:
After discussion with @maximearmstrong we could in a first step implement @barbeau  solution: check values (using getters on the builder) in the usecase and add notices for the relevant cases, and in a second step refactor Notices.

Of course, this depends on our priorities

Reply:
Implementation comes in https://github.com/MobilityData/gtfs-validator/pull/126